pkg/systemd/services/service-details.jsx

```diff
@@ -208,11 +209,71 @@ export class ServiceDetails extends React.Component {
             waitsFileAction: false,
             note: "",
             error: "",
+            unit_properties: {},
         };
 
         this.onOnOffSwitch = this.onOnOffSwitch.bind(this);
         this.unitAction = this.unitAction.bind(this);
         this.unitFileAction = this.unitFileAction.bind(this);
+
+        this.unitType = props.unit.Id.split('.').slice(-1)[0];
+        this.unitTypeCapitalized = this.unitType.charAt(0).toUpperCase() + this.unitType.slice(1);
+        this.doMemoryCurrentPolling = this.doMemoryCurrentPolling.bind(this);
+
+        // Subscribe to PropertiesChanged - follow changes on Listen property
+        if (this.unitType == 'socket') {
+            systemd_client.subscribe(
+                {
+                    path: props.unit.path,
+                    interface: "org.freedesktop.DBus.Properties",
+                    member: "PropertiesChanged"
+                },
+                (path, iface, signal, args) => {
+                    if (args[1].Listen) {
```

Comment:
It still amazes me how a socket unit could change its `ListenStream` property while it's running. Did you observe that in practice? How would one do that?

Reply:
No I did not, I just saw it in the API documentation that it emits signal when changed. Also in your previous review I thought you wanted me to write a event listener to follow changes for this property, but I guess I misunderstood the comment. Shall I just remove it then? I also feel it's useless to have an even listener for this.

Reply:
Sorry for the misunderstanding! Yes, please remove it all; it can't change while a .socket unit is running.

Reply:
So I moved the 'LIsten' property fetching one component up, because where it was we would not receive updates even if the unit restarted.